### PR TITLE
Morpho table

### PR DESCRIPTION
**Pull request by scseeman**

### CODE REVIEW

File: multipatch_analysis/database/morphology.py [hunk 1]
```diff
@@ -7,7 +7,18 @@
 
 Morphology = make_table(name='morphology', comment="Describes morphological properties of cells.", columns=[
     ('cell_id', 'cell.id', 'The ID of the cell described by each record', {'index': True, 'unique': True}),
-    ('pyramidal', 'bool', 'Whether the experimenter labeled this cell as pyramidal', {'index': True}),
+    ('cell_specimen_id', 'int', 'LIMS cell specimen ID', {'index': True}),
```
Comment (campagnola) [inline multipatch_analysis/database/morphology.py]: Why's this here? If anything it should be in the cell table

File: multipatch_analysis/database/morphology.py [hunk 2]
```diff
@@ -7,7 +7,18 @@
...
+    ('morpho_db_hash', 'int', 'hash of Morphology team database for update tracking', {'index': True}),
```
Comment (campagnola) [inline multipatch_analysis/database/morphology.py]: No index here.

File: multipatch_analysis/pipeline/morphology.py [hunk 3]
```diff
@@ -44,7 +70,32 @@ def create_db_entries(cls, job_id, session):
...
+
             }
+            
+            if cell_morpho is not None:
+                morpho_db_hash = hash(';'.join(filter(None, cell_morpho)))
+                results['morpho_db_hash'] = morpho_db_hash
+                for morpho_db_name, result in col_names.items():
+                    col_name = result['name']
+                    col_type = result['type']
+                    data = cell_morpho.__getattribute__(morpho_db_name)
```
Comment (campagnola) [inline multipatch_analysis/pipeline/morphology.py]: Preferred: `getattr(cell_morpho, morpho_db_name)`

File: multipatch_analysis/pipeline/morphology.py [hunk 4]
```diff
@@ -65,20 +116,57 @@ def ready_jobs(self):
...
+    cnxn_str = r'DRIVER={Microsoft Access Driver (*.mdb, *.accdb)}; DBQ=%s' % config.morpho_address
+    cnxn = pyodbc.connect(cnxn_str)
+    cursor = cnxn.cursor()
+    morpho_table = cursor.execute('select * from MPATCH_CellsofCluster')
+    results = morpho_table.fetchall()
+    morpho_results = {int(r.cell_specimen_id): r for r in results}
+
+    return morpho_results
+
+def get_expt_cell_cluster(expt):
```
Comment (campagnola) [inline multipatch_analysis/pipeline/morphology.py]: Seems like something that could be in `lims.py` or a method of `Experiment` ?

File: multipatch_analysis/database/morphology.py (hunk 1 above)
Comment (scseeman) [reply multipatch_analysis/database/morphology.py]: sure we can put it in the cell table.....this is how morphology is referencing everything so, a) I needed to generate it and b) I didn't want to keep having to do that

File: multipatch_analysis/database/morphology.py (hunk 2 above)
Comment (scseeman) [reply multipatch_analysis/database/morphology.py]: what does the index do exactly?

File: multipatch_analysis/pipeline/morphology.py (hunk 3 above)
Comment (scseeman) [reply multipatch_analysis/pipeline/morphology.py]: for some reason I feel like I couldn't get that to work, but let me try again

File: multipatch_analysis/pipeline/morphology.py (hunk 4 above)
Comment (scseeman) [reply multipatch_analysis/pipeline/morphology.py]: or stored in the Experiment table as it's supposed to be :)

File: multipatch_analysis/database/morphology.py (hunk 2 above)
Comment (campagnola) [reply multipatch_analysis/database/morphology.py]: makes the column faster to search (which we don't need) and slower to write.

**Comment by scseeman** [comment]
@campagnola this is ready to go from my perspective

File: tools/matrix_analyzer.py [hunk 5]
```diff
@@ -17,55 +17,56 @@
...
+            {'cre_type': 'sst', 'target_layer': '2/3', 'cortical_layer': '2/3'},
+            {'cre_type': 'vip', 'target_layer': '2/3', 'cortical_layer': '2/3'},
         ]),
         
         ('Mouse Layer 4', [
-            {'cre_type': 'nr5a1', 'target_layer': '4'},
-            {'cre_type': 'pvalb', 'target_layer': '4'},
-            {'cre_type': 'sst', 'target_layer': '4'},
-            {'cre_type': 'vip', 'target_layer': '4'},
+            {'cre_type': 'nr5a1', 'target_layer': '4', 'cortical_layer': '4'},
```
Comment (campagnola) [inline tools/matrix_analyzer.py]: Why use both target_layer and cortical_layer?

Comment (scseeman) [reply tools/matrix_analyzer.py]: because sometimes you want to define by the target layer and other times by the annotated layer

Comment (campagnola) [reply tools/matrix_analyzer.py]: But I think this would make the cell class require _both_ to be true.

Comment (scseeman) [reply tools/matrix_analyzer.py]: I think I edited cell_class to make it an either or situation, but maybe we should look at it? When I tested it, the output seemed reasonable
